### PR TITLE
Add azure-cli tool

### PR DESCRIPTION
**Pull request by nodauf**
# Description

Add [azure-cli](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli-linux) tool. This is a tool made by Microsoft to connect to Azure and execute administrative commands on Azure resources.

This tool can be used to detect misconfiguration about authentication (like shown recently https://twitter.com/shirtamari/status/1643300698044391466 and https://www.wiz.io/blog/azure-active-directory-bing-misconfiguration)

### CODE REVIEW

File: sources/install.sh [hunk 1]
```diff
@@ -1639,6 +1639,12 @@ function install_scout() {
   add-test-command "scout --help"
 }
 
+function install_azurecli(){
+  colorecho "Installing Azure cli"
+  fapt azure-cli
+  add-test-command "az --help"
+}
```
Comment (ShutdownRepo) [inline sources/install.sh:1646]: Could you add `add-history` if you have some example commands in mind?

**Review by ShutdownRepo — CHANGES_REQUESTED** [review]

File: sources/install.sh [hunk 2]
```diff
@@ -1639,6 +1639,12 @@ function install_scout() {
   add-test-command "scout --help"
 }
 
+function install_azurecli(){
+  colorecho "Installing Azure cli"
+  fapt azure-cli
+  add-test-command "az --help"
```
Comment (QU35T-code) [inline sources/install.sh:1645]: The best command would be : `az version` in order to be compliant with other tools.

Comment (nodauf) [reply sources/install.sh]: The test command has been updated to `az version`

**Review by QU35T-code — APPROVED** [review]

File: sources/install.sh (hunk 1 above)
Comment (nodauf) [reply sources/install.sh:1647]: I am not used to this tool, so I add only one command. The one that is used in the Twitter thread and in the article to detect multi-tenant authentication.
`az ad sp show --id $TENANT_ID`

**Comment by ShutdownRepo** [comment]
A refactor is in progress ( https://github.com/ThePorgs/Exegol-images/pull/137 ) and should be over with shortly. Putting this PR on hold until then.
Once the refactor is over with, we will be able to update this PR and merge it 👍
Sorry for the inconvenience

**Review by ShutdownRepo — APPROVED** [review]

**Review by QU35T-code — APPROVED** [review]

**Comment by ShutdownRepo** [comment]
@nodauf refactor is done. Changes were conducted on the `dev` branch. Can you sync and change your PR according to the new structure?
1. sync the latest changed on our `dev` branch to your branch
2. move your changes where they're supposed to end up at
If that's easier for you, you can close this PR and create another one if you'd like

**Pull request closed without merging**